### PR TITLE
File/line should point to the caller, not to log2logstash

### PR DESCRIPTION
**Pull request by sjinks**
## Changelog Description

Because of the way how `Logger::parse_params()` retrieved the backtrace, `file` and `line` always pointed to `Logger::log2logstash()`.

This PR fixes that; `file`/`line` will now point to the location that invoked `Logger::log2logstash()`; in case the method was invoked from the `log2logstash()` function, `file`/`line` will point to the caller of `log2logstash()`.

### Plugin Updated: VIP Init

`file` and `line` parameters of the payload will now point to the caller of `log2logstash()` function/method.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

### CODE REVIEW

**Review by jesse-greathouse — APPROVED** [review]
Good logic but hard to follow. I know it depends on the data format, of the object graph its analyzing; its just the PHP is hard to grok.

Could use 2 or 3 comment lines to tell the reader what the logic is supposed to do; to prime the reader.

**Comment by sjinks** [comment]
r2296-stacks

**Comment by sjinks** [comment]
v20220216.0
